velox/serializers/PrestoSerializer.cpp

```diff
@@ -1333,6 +1423,69 @@ void serializeColumn(
   }
 }
 
+template <TypeKind Kind>
+void serializeConstantColumn(
+    const BaseVector* vector,
+    const folly::Range<const IndexRange*>& ranges,
+    VectorStream* stream) {
+  using T = typename KindToFlatVector<Kind>::WrapperType;
+
+  for (const auto& range : ranges) {
+    stream->appendNonNull(range.size);
+  }
+
+  std::vector<IndexRange> newRanges;
+  newRanges.push_back({0, 1});
+  serializeConstantVector<Kind>(vector, newRanges, stream->childAt(0));
+}
+
+template <TypeKind Kind>
+void serializeDictionaryColumn(
+    const BaseVector* vector,
+    const folly::Range<const IndexRange*>& ranges,
+    VectorStream* stream) {
+  using T = typename KindToFlatVector<Kind>::WrapperType;
+
+  auto dictionaryVector = dynamic_cast<const DictionaryVector<T>*>(vector);
+  VELOX_CHECK_NULL(
```

Comment:
do we have logic somewhere up the stack to check if this has nulls and fallback to flat serialization instead?

Reply:
No. This is test-only API and I want it to actually produce dictionary encoding. The test verifies that dictionary encoding is preserved across serde round trip.